src/main/scala/com.snowplowanalytics.sauna/apis/OpsGenie.scala

```diff
@@ -0,0 +1,166 @@
+
+// sauna
+import loggers.Logger.Notification
+import utils._
+
+/**
+ * Opsgenie API wrapper. Encapsulates all communications with Opsgenie
+ *
+ * @param apiKey OpsGenie token.
+ * @param logger A logger actor.
+ */
+class OpsGenie(apiKey: String, logger: ActorRef) {
+  import OpsGenie._
+
+  val client = makeClient(apiKey)
+
+  /**
+   * Create alert request is used to create alerts in OpsGenie
+   *
+   * @param  alert  OpsGenie Alert object
+   *
+   * @return Future[CreateAlertResponse]
+   */
+  def createAlert(alert: Alert): Future[CreateAlertResponse] = Future{
+    CreateAlertRequest(alert)(client).toResponse
+  }
+
+}
+
+object OpsGenie {
+  import com.ifountain.opsgenie.client.OpsGenieClient
+  import com.ifountain.opsgenie.client.swagger.model.{CreateAlertRequest => POJOCreateAlertRequest, TeamRecipient}
+  import com.ifountain.opsgenie.client.swagger.model.{SuccessResponse => POJOSuccessResponse}
+  import scala.util.{Try, Success, Failure}
+  import scala.language.postfixOps
+  
+  type Alias = String
+  type Description = String
+  type Tag = String
+  type Recipient = String
+  type User = String
+  type Note = String
+  type Action = String
+  type Message = String
+  type Entity = String
+  type Source = String
+  type Team = String
+
+  def makeClient(apiKey: String) = {
+    val client = new OpsGenieClient()
+    client.setApiKey(apiKey)
+    client
+  }
+
+  case class Alert(
+    val message:     Message,
+    val teams:       Option[Seq[Team]] = None,
+    val alias:       Option[Alias] = None,
+    val description: Option[Description] = None,
+    val recipients:  Option[Seq[Recipient]] = None,
+    val actions:     Option[Seq[Action]] = None,
+    val source:      Option[Source] = None,
+    val tags:        Option[Seq[Tag]] = None,
+    val details:     Option[Map[String, String]] = None,
+    val entity:      Option[Entity] = None,
+    val user:        Option[User] = None,
+    val note:        Option[Note] = None){
+
+    lazy val request: POJOCreateAlertRequest = {
+      val request: POJOCreateAlertRequest =  new POJOCreateAlertRequest();
+      request.setMessage(message)
+      if(alias.isDefined) request.setAlias(alias.get)
+      if(description.isDefined) request.setDescription(description.get)
+      if(entity.isDefined) request.setEntity(entity.get)
+      if(user.isDefined) request.setUser(user.get)
+      if(note.isDefined) request.setNote(note.get)
+        
+      if(teams.isDefined) teams.get.foreach{t => request.addTeamsItem(new TeamRecipient().name(t))}
+      if(recipients.isDefined) recipients.get.foreach{r => request.addTeamsItem(new TeamRecipient().name(r))}
+      if(actions.isDefined) actions.get.foreach{a => request.addActionsItem(a)}
+      if(tags.isDefined) tags.get.foreach{t => request.addTagsItem(t)}
+      
+      request  
+    }
+  }
+
+  /**
+   * Custom reader for an Alert instance.
+   */
+  implicit val alertReads: Reads[Alert] = (
+      (JsPath \ "message").read[Message] and
```

Comment:
From https://docs.opsgenie.com/docs/web-api/alert-api#createAlertRequest ,message is mandatory